timescaledb/tutorials/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,347 @@
+    active until it gets terminated. If you don't add this line the connection
+    might break instantly.
+
+1. Now run the Python script:
+    ```bash
+    python websocket_test.py
+    ```
+1. After running the script, you immediately get a response from the server 
+    about the status of your connection:
+    ```bash
+    {'event': 'subscribe-status',
+     'status': 'ok',
+     'success': [
+            {'symbol': 'BTC/USD', 'exchange': 'Coinbase Pro', 'mic_code': 'Coinbase Pro', 'country': '', 'type': 'Digital Currency'},
+            {'symbol': 'ETH/USD', 'exchange': 'Huobi', 'mic_code': 'Huobi', 'country': '', 'type': 'Digital Currency'}
+        ],
+     'fails': None
+    }
+    ```
+
+</procedure>
+
+Then, if you wait a few seconds you see actual data records printed out:
+
+```bash
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438893, 'price': 30361.2, 'bid': 30361.2, 'ask': 30361.2, 'day_volume': 49153}
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438896, 'price': 30380.6, 'bid': 30380.6, 'ask': 30380.6, 'day_volume': 49157}
+{'event': 'heartbeat', 'status': 'ok'}
+{'event': 'price', 'symbol': 'ETH/USD', 'currency_base': 'Ethereum', 'currency_quote': 'US Dollar', 'exchange': 'Huobi', 'type': 'Digital Currency', 'timestamp': 1652438899, 'price': 2089.07, 'bid': 2089.02, 'ask': 2089.03, 'day_volume': 193818}
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438900, 'price': 30346.0, 'bid': 30346.0, 'ask': 30346.0, 'day_volume': 49167}
+```
+
+Each price event gives you multiple data points about the given trading pair.
+For example, the name of the exchange, current price, etc... You will
+occasionally see `heartbeat` events in the response - these events signal
+the health of the connection over time.
+
+At this point the websocket connection works and data keeps flowing. You need
+to implement the `on_event` function so data gets ingested into TimescaleDB.
+
+## Ingesting websocket data into TimescaleDB
+
+Now that the websocket connection is set up, implement the `on_event` function
+so data gets inserted into the database. When ingestin data into a
+transactional database, generally speaking, ingesting a thousand records in
+one transaction will be faster than ingesting them in a thousand transactions.
+
+TimescaleDB, being a transactional database, creates a new transaction
+whenever you try to insert a new record. Because of this behavior, it's suggested to
+insert data records in a batch - multiple records in one transaction.
+Using this approach you can achieve higher ingest rates than if you were to
+insert data record-by-record.
```

Comment:
```suggestion
Because TimescaleDB is a transactional database, it creates a new transaction
whenever you try to insert a new record. This means that it's better to insert 
data records in a batch, so you can ingest multiple records in a single transaction.
This approach means you can achieve higher ingest rates than if you were to
insert data records one-by-one.
```